src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts

```diff
@@ -0,0 +1,259 @@
+import { Component, EventEmitter, Input, OnInit, Output } from "@angular/core";
+import { FormControl, FormGroup, Validators } from "@angular/forms";
+import { CountryISO } from "ngx-intl-tel-input";
+import { GlobalVarsService } from "../global-vars.service";
+import { BackendAPIService, User } from "../backend-api.service";
+import { ActivatedRoute } from "@angular/router";
+import { IdentityService } from "../identity.service";
+import { AccountService } from "../account.service";
+
+@Component({
+  selector: "sign-up-get-starter-deso",
+  templateUrl: "./sign-up-get-starter-deso.component.html",
+  styleUrls: ["./sign-up-get-starter-deso.component.scss"],
+})
+export class SignUpGetStarterDESOComponent implements OnInit {
+  static CREATE_PHONE_NUMBER_VERIFICATION_SCREEN = "create_phone_number_verification_screen";
+  static SUBMIT_PHONE_NUMBER_VERIFICATION_SCREEN = "submit_phone_number_verification_screen";
+  static COMPLETED_PHONE_NUMBER_VERIFICATION_SCREEN = "completed_phone_number_verification_screen";
+
+  @Input() displayForSignupFlow: boolean = false;
+  publicKey: string = "";
+  @Output() backToPreviousSignupStepClicked = new EventEmitter();
+  @Output() phoneNumberVerified = new EventEmitter();
+  @Output() skipButtonClicked = new EventEmitter();
+
+  phoneForm = new FormGroup({
+    phone: new FormControl(undefined, [Validators.required]),
+  });
+  verificationCodeForm = new FormGroup({
+    verificationCode: new FormControl(undefined, [Validators.required]),
+  });
+
+  CountryISO = CountryISO;
+  sendingPhoneNumberVerificationText = false;
+  submittingPhoneNumberVerificationCode = false;
+  screenToShow: string | null = null;
+  SignUpGetStarterDESOComponent = SignUpGetStarterDESOComponent;
+  phoneNumber: string = "";
+  phoneNumberCountryCode: string | null = null;
+  resentVerificationCode = false;
+  sendPhoneNumberVerificationTextServerErrors = new SendPhoneNumberVerificationTextServerErrors();
+  submitPhoneNumberVerificationCodeServerErrors = new SubmitPhoneNumberVerificationCodeServerErrors();
+  user: User | null = null;
+  loading: boolean = true;
+  isPhoneNumberSuccess: boolean = false;
+
+  constructor(
+    public globalVars: GlobalVarsService,
+    private backendApi: BackendAPIService,
+    private activatedRoute: ActivatedRoute,
+    private identityService: IdentityService,
+    private accountService: AccountService,
+  ) {}
+
+  ngOnInit(): void {
+    this._setScreenToShow();
+    this.activatedRoute.queryParams.subscribe(params => {
+      if (params.public_key) {
+        this.publicKey = params.public_key;
+        this.backendApi.GetUsersStateless([this.publicKey]).subscribe((res) => {
+          if (res.UserList?.length) {
+            this.user = res.UserList[0];
+          }
+        }, (err) => {
+          console.error(err);
+        }).add(() => this.loading = false);
+      } else {
+        this.loading = false;
+      }
+    });
+
+  }
+
+  _setScreenToShow() {
```

Comment:
Why is this necessary? Why not just put this in the GetUsersStateless callback?

Reply:
Yeah I should move this. I just copied this from the code that was in frontend.